docs/architecture/ADR-005-QGB_reduce_state_usage.md

```diff
@@ -0,0 +1,120 @@
+Then, keep only the state that is created in [EndBlocker](https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/x/qgb/abci.go#L12-L16).
+Which are `Attestations`, i.e. `Valset`s and `DataCommitmentRequest`s.
+
+### QGB Rollup
+
+Deploy the QGB as a Rollup that posts its data to Celestia, and, uses a separate settlement layer for slashing.
+This might be the end goal of the QGB, but it will be very involved to build at this stage.
+
+Also, if this ADR got accepted, it will be an important stepping stone in the Rollup direction.
+
+## Decision
+
+We will need to decide on two things:
+
+- [ ] Should we go for this approach?
+- [ ] Should this change be part of QGB 1.0?
+- [ ] When do we want to implement slashing?
+
+## Detailed Design
+
+The proposed design consists of keeping the same transaction types we currently have : the `MsgValsetConfirm` defined in [here](https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/proto/qgb/msgs.proto#L24-L49), and the `MsgDataCommitmentConfirm` defined in [here](
+https://github.com/celestiaorg/celestia-app/blob/a965914b8a467f0384b17d9a8a0bb1ac62f384db/proto/qgb/msgs.proto#L55-L76). However, remove  all the message server checks defined in the [msg_server.go](https://github.com/celestiaorg/celestia-app/blob/qgb-integration/x/qgb/keeper/msg_server.go) :
+
+```go
+// ValsetConfirm handles MsgValsetConfirm.
+func (k msgServer) ValsetConfirm(
+    c context.Context,
+    msg *types.MsgValsetConfirm,
+) (*types.MsgValsetConfirmResponse, error) {
+    // <delete_all_this>
+}
+
+// DataCommitmentConfirm handles MsgDataCommitmentConfirm.
+func (k msgServer) DataCommitmentConfirm(
+    c context.Context,
+    msg *types.MsgDataCommitmentConfirm,
+) (*types.MsgDataCommitmentConfirmResponse, error) {
+    // <delete_all_this>
+}
+```
+
+This would reduce significantly the QGB module state usage, reduce the complexity of the state machine and give us the same benefits of the current design.
+
+As a direct consequence to this, the relayer will need to adapt to this change and start getting the transactions straight from the blocks, parse them and sanitize the commits. Thus, making the relayer implementation more complex.
+
+However, we can assume that the relayer will have enough computing power to do the latter. Also, only one relayer is necessary to have a working QGB contract. So, the relayer cost is justified.
+
+For the orchestrators, they will also need to parse the history to keep track of any missed signatures. But, same as with the relayers, the cost is justified.
+
+For posting transactions, we will rely on gas fees as a mechanism to limit malicious parties to flood the network with invalid transactions. Then, eventually slash malicious behavious. However, since posting confirms will be possible for any user of the network. It won't be possible to slash ordinary users, who are not running validators, if they post invalid confirms.
```

Comment:
> If we assume that the QGB data (i.e. Valset Confirms and DataCommitment Confirms) is stored in a non-reserved namespace, I think the problem stated above == https://forum.celestia.org/t/woods-attack-on-celestia/59

We are no longer posting in confirms in namespaces, so woods attack is not relevant, if I understand right.

> If the QGB data is only stored in transaction data and this document proposed adding new transaction types (distinct from MsgPayForData), I think we need a follow-up issue to discuss the metering (i.e. gas accounting) for these new message types.

No need for a new message type, we already have them defined and used. And the gas usage will be minimal since there will be no read/write in state.